main.go

```diff
@@ -168,7 +176,17 @@ func realMain() error {
 
 func (c *config) rewrite() (ast.Node, error) {
 	c.fset = token.NewFileSet()
-	node, err := parser.ParseFile(c.fset, c.file, nil, parser.ParseComments)
+	var contents interface{}
+	if c.modified != nil {
+		archive, err := buildutil.ParseOverlayArchive(c.modified)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse -modified archive: %v", err)
+		}
+		if fc, ok := archive[c.file]; ok {
+			contents = fc
```

Comment:
What if `c.file` doesn't exist here? Can you also return error in that case and continue